plan/logical_plan_builder.go

```diff
@@ -409,6 +409,11 @@ func (b *planBuilder) buildSelection(p LogicalPlan, where ast.ExprNode, AggMappe
 				ret, err := expression.EvalBool(expression.CNFExprs{con}, nil, b.ctx)
 				if ret || err != nil {
 					continue
+				} else {
```

Comment:
else if err == nil.

if err != nil , maybe there is a runtime error. Let it report in executor.